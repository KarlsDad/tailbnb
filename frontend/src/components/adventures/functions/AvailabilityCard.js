import React from 'react';

export default () => {
  return (
    <div className='w-full flex justify-start items-start flex-wrap'>
      <div className='w-23p bg-white shadow-md rounded-xl mr-4'>
        <div className='m-8'>
          <h3
            style={{ fontFamily: 'airbnb-medium' }}
            className='text-gray-750 text-2xl py-1'>
            Mar 03
          </h3>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='text-gray-750 text-lg'>
            Tuesday
          </p>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='text-gray-750 text-lg'>
            From $130/person
          </p>
          <button
            style={{ fontFamily: 'airbnb-bold' }}
            className='text-green-850 mt-8'>
            Book
          </button>
        </div>
      </div>
    </div>
  );
};
