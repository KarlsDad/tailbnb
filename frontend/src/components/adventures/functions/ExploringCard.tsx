import * as React from 'react';

interface Props {
  title: string;
  experienceNumber: number;
}

export const ExploringCard: React.FC<Props> = ({ title, experienceNumber }) => {
  return (
    <div className='bg-white shadow-sm rounded mr-4 border border-gray-300 mt-4'>
      <div className='px-5 py-5 mr-5'>
        <h3 style={{ fontFamily: 'airbnb-bold' }} className='text-gray-750'>
          {title}
        </h3>
        <p
          style={{ fontFamily: 'airbnb-medium' }}
          className='text-xs text-gray-650'>
          {experienceNumber} experiences
        </p>
      </div>
    </div>
  );
};
