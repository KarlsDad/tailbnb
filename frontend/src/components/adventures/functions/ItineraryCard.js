import React from 'react';

// Images
const detail2 = require('../../../../public/img/high/staydetail2.jpg');

export default () => {
  return (
    <div
      style={{ fontFamily: 'airbnb-book' }}
      className='px-5 text-gray-750 border-l border-gray-800 relative z-10'>
      <div style={{ marginBottom: 6 }}>
        <div
          style={{
            height: 7,
            width: 7,
            top: -7,
            left: -3.6
          }}
          className='absolute z-50'>
          <svg
            className='h-full w-full'
            xmlns='http://www.w3.org/2000/svg'
            viewBox='0 0 512 512'>
            <g>
              <g>
                <path
                  d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                />
              </g>
            </g>
          </svg>
        </div>

        <p className='mb-2'>Day 1</p>
        <div className='w-full md:flex justify-start items-start pb-8'>
          <div className='md:w-5/12'>
            <img className='h-full w-full' src={detail2} alt='' />
          </div>

          <div className='w-full md:w-7/12 md:pl-5'>
            <p
              style={{ fontFamily: 'airbnb-bold' }}
              className='pt-3 pb-2 md:pt-0 md:pb-0 text-base'>
              Marrakech-Oughla - Tinzerte
            </p>
            <p className='text-base'>
              Marrakech to imi oughlad small village situated 1,450m at the base
              of Toubkal. Here we begin the gentle accent.
            </p>
            <button
              style={{ fontFamily: 'airbnb-bold' }}
              className='text-green-850 text-sm mt-4'>
              Show more
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

{
  /* <div
        style={{ height: 7, width: 7, bottom: -7, left: -3.6 }}
        className='absolute z-100'>
        <svg
          className='h-full w-full'
          xmlns='http://www.w3.org/2000/svg'
          viewBox='0 0 512 512'>
          <g>
            <g>
              <path
                d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
              />
            </g>
          </g>
        </svg>
      </div> */
}
