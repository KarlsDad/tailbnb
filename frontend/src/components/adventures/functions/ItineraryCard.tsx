import * as React from 'react';

// Images
const detail2 = require('../../../../public/img/high/staydetail2.jpg');

interface Props {
  is_last: boolean;
  day: number;
  title: string;
  description: string;
}

export const ItineraryCard: React.FC<Props> = ({
  is_last,
  day,
  title,
  description
}) => {
  return (
    <>
      {is_last ? (
        <div
          style={{ fontFamily: 'airbnb-book' }}
          className='px-5 text-gray-750 relative'>
          <div style={{ marginBottom: 6 }}>
            <div
              style={{
                height: 7,
                width: 7,
                top: -6.5,
                left: -3.4
              }}
              className='absolute z-50'>
              <svg
                className='h-full w-full'
                xmlns='http://www.w3.org/2000/svg'
                viewBox='0 0 512 512'>
                <g>
                  <g>
                    <path
                      d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                    />
                  </g>
                </g>
              </svg>
            </div>

            <p className='mb-2'>Day {day}</p>
            <div className='w-full md:flex justify-start items-start pb-8'>
              <div className='md:w-5/12'>
                <img className='h-full w-full' src={detail2} alt='' />
              </div>

              <div className='w-full md:w-7/12 md:pl-5'>
                <p
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='pt-3 pb-2 md:pt-0 md:pb-0 text-base'>
                  {title}
                </p>
                <p className='text-base'>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='text-green-850 text-sm mt-4'>
                  Show more
                </button>
              </div>
            </div>
          </div>
        </div>
      ) : (
        <div
          style={{ fontFamily: 'airbnb-book' }}
          className='px-5 text-gray-750 border-l border-gray-800 relative'>
          <div style={{ marginBottom: 6 }}>
            <div
              style={{
                height: 7,
                width: 7,
                top: -7,
                left: -3.6
              }}
              className='absolute'>
              <svg
                className='h-full w-full'
                xmlns='http://www.w3.org/2000/svg'
                viewBox='0 0 512 512'>
                <g>
                  <g>
                    <path
                      d='M256,0C115.03,0,0,115.05,0,256c0,140.97,115.05,256,256,256c140.97,0,256-115.05,256-256C512,115.03,396.95,0,256,0z
			 M256,482C131.383,482,30,380.617,30,256S131.383,30,256,30s226,101.383,226,226S380.617,482,256,482z'
                    />
                  </g>
                </g>
              </svg>
            </div>

            <p className='mb-2'>Day {day}</p>
            <div className='w-full md:flex justify-start items-start pb-8'>
              <div className='md:w-5/12'>
                <img className='h-full w-full' src={detail2} alt='' />
              </div>

              <div className='w-full md:w-7/12 md:pl-5'>
                <p
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='pt-3 pb-2 md:pt-0 md:pb-0 text-base'>
                  {title}
                </p>
                <p className='text-base'>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-bold' }}
                  className='text-green-850 text-sm mt-4'>
                  Show more
                </button>
              </div>
            </div>
          </div>
        </div>
      )}
    </>
  );
};
