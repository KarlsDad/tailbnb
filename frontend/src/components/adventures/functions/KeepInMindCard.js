import React from 'react';

export default ({ title, description, order }) => {
  return (
    <>
      {order === 'top' ? (
        <div className='w-full flex pb-4 border-b border-gray-300'>
          <div className='w-1/3'>
            <h3 style={{ fontFamily: 'airbnb-medium' }} className='text-2xl'>
              {title}
            </h3>
          </div>
          <div className='w-2/3'>
            <p style={{ fontFamily: 'airbnb-book' }}>{description}</p>
            <button
              style={{ fontFamily: 'airbnb-medium' }}
              className='text-green-850 text-sm my-3'>
              Learn more
            </button>
          </div>
        </div>
      ) : (
        <>
          {order === 'bottom' ? (
            <div className='w-full flex pb-4 pt-6'>
              <div className='w-1/3'>
                <h3
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-2xl'>
                  {title}
                </h3>
              </div>
              <div className='w-2/3'>
                <p style={{ fontFamily: 'airbnb-book' }}>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-green-850 text-sm my-3'>
                  Learn more
                </button>
              </div>
            </div>
          ) : (
            <div className='w-full flex pb-4 pt-6 border-b border-gray-300'>
              <div className='w-1/3'>
                <h3
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-2xl'>
                  {title}
                </h3>
              </div>
              <div className='w-2/3'>
                <p style={{ fontFamily: 'airbnb-book' }}>{description}</p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-green-850 text-sm my-3'>
                  Learn more
                </button>
              </div>
            </div>
          )}
        </>
      )}
    </>
  );
};
