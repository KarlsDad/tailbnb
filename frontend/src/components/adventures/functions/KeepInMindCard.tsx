import * as React from 'react';

interface Props {
  title: string;
  description: string;
  order?: string | undefined;
}

export const KeepInMindCard: React.FC<Props> = ({
  title,
  description,
  order
}) => {
  return (
    <>
      {order === 'top' ? (
        <div className='w-full flex pb-6 pt-6'>
          <div className='w-full md:w-1/3 h-full items-center'>
            <h3
              style={{ fontFamily: 'airbnb-medium' }}
              className='pt-2 text-lg lg:text-2xl'>
              {title}
            </h3>
          </div>

          <div className='w-2/3 pl-4'>
            <p
              className='hidden lg:block'
              style={{ fontFamily: 'airbnb-book' }}>
              {description}
            </p>
            <button
              style={{ fontFamily: 'airbnb-medium' }}
              className='hidden lg:block text-green-850 text-sm my-3'>
              Learn more
            </button>
          </div>
        </div>
      ) : (
        <>
          {order === 'bottom' ? (
            <div className='w-full flex pb-6 pt-6'>
              <div className='w-1/3'>
                <h3
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-lg lg:text-2xl'>
                  {title}
                </h3>
              </div>
              <div className='w-2/3 pl-4'>
                <p
                  className='hidden lg:block'
                  style={{ fontFamily: 'airbnb-book' }}>
                  {description}
                </p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='hidden lg:block text-green-850 text-sm my-3'>
                  Learn more
                </button>
              </div>
            </div>
          ) : (
            <div className='w-full flex pb-6 pt-6'>
              <div className='w-1/3'>
                <h3
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='text-lg lg:text-2xl'>
                  {title}
                </h3>
              </div>
              <div className='w-2/3 pl-4'>
                <p
                  className='hidden lg:block'
                  style={{ fontFamily: 'airbnb-book' }}>
                  {description}
                </p>
                <button
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='hidden lg:block text-green-850 text-sm my-3'>
                  Learn more
                </button>
              </div>
            </div>
          )}
        </>
      )}
    </>
  );
};
