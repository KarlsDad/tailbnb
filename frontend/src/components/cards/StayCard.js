import React, { useState } from 'react';
import LazyImage from 'react-lazy-progressive-image';
import Link from 'next/link';

export default ({
  id,
  hostType,
  img,
  imglow,
  location,
  title,
  cost,
  ratings
}) => {
  const renderhost = () => {
    if (hostType === 'normal') {
      return (
        <div className='md:flex md:flex-wrap md:items-center md:justify-start'>
          <p className='mt-3 text-sm sm:my-2 font-light text-gray-600'>
            {location}
          </p>
        </div>
      );
    } else if (hostType === 'superhost') {
      return (
        <div className='md:flex md:flex-wrap md:items-center md:justify-start'>
          <p className='uppercase mt-3 md:mt-0 border border-gray-800 rounded font-semibold px-1 mt-1 text-xs'>
            {hostType}
          </p>
          <p className='sm:ml-2 mt-1 text-sm sm:my-2 font-light text-gray-600'>
            {location}
          </p>
        </div>
      );
    } else if (hostType === 'plus') {
      return (
        <div className='sm:flex sm:flex-wrap sm:items-center sm:justify-start'>
          <p className='inline-block mt-3 md:mt-0 px-1 rounded text-white uppercase tracking-wide text-xs font-semibold bg-pink-800'>
            {hostType}
          </p>
          <p className='sm:ml-2 mt-1 text-sm sm:my-2 font-light text-gray-600'>
            {location}
          </p>
        </div>
      );
    }
  };

  return (
    <Link href='/rooms/[id]' as={`/rooms/${id}`}>
      <div className='w-30/31'>
        <LazyImage src={img} placeholder={imglow}>
          {(src, loading, isVisible) => (
            <img className='w-full rounded' src={src} />
          )}
        </LazyImage>

        <div className='flex flex-wrap items-center justify-between'>
          {renderhost()}
          <div className='flex items-center justify-center flex-wrap'>
            <svg
              xmlns='http://www.w3.org/2000/svg'
              x='0px'
              y='0px'
              width='13'
              height='13'
              viewBox='0 0 37.286 37.287'
              style={{ fill: '#F5385D' }}>
              <g>
                <path
                  d='M36.683,16.339l-7.567,7.377l1.786,10.417c0.128,0.75-0.182,1.509-0.797,1.957c-0.348,0.253-0.762,0.382-1.176,0.382
		c-0.318,0-0.638-0.076-0.931-0.23l-9.355-4.918l-9.355,4.918c-0.674,0.355-1.49,0.295-2.107-0.15
		c-0.615-0.448-0.924-1.206-0.795-1.957l1.787-10.417L0.604,16.34c-0.547-0.531-0.741-1.326-0.508-2.05
		c0.236-0.724,0.861-1.251,1.615-1.361l10.459-1.521l4.68-9.478c0.335-0.684,1.031-1.116,1.792-1.116
		c0.763,0,1.456,0.432,1.793,1.115l4.68,9.478l10.461,1.521c0.752,0.109,1.379,0.637,1.611,1.361
		C37.425,15.013,37.226,15.808,36.683,16.339z'
                />
              </g>
            </svg>
            <p className='pl-1 text-sm'>{ratings}</p>
          </div>
        </div>
        <p className='my-1'>{title}</p>
        <p className='tracking-wide'>
          <span className='font-bold'>${cost} CAD</span>/night
        </p>
      </div>
    </Link>
  );
};
