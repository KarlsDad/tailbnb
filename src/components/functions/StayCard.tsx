import * as React from 'react';
import Link from 'next/link';

interface Prop {
  id: string;
  host_is_superhost: string;
  name: string;
  price: number;
  country: string;
  reviews_per_month: number;
  picture_url: string;
}

export const StayCard: React.FC<Prop> = ({
  id,
  host_is_superhost,
  name,
  price,
  country,
  reviews_per_month,
  picture_url
}) => {
  const renderhost = (host_is_superhost: string) => {
    if (host_is_superhost == 't') {
      return (
        <>
          <div className='flex items-center justify-between'>
            <div className='flex flex-wrap items-center'>
              <div className='mt-2 mb-1'>
                <p
                  style={{ fontFamily: 'airbnb-medium' }}
                  className='uppercase border border-gray-800 rounded text-xs px-1'>
                  Superhost
                </p>
              </div>
              <div className='ml-1 mt-1'>
                <p
                  style={{ fontFamily: 'airbnb-book' }}
                  className='text-sm text-gray-750'>
                  {country}
                </p>
              </div>
            </div>
            <div className='flex'>
              <div className='h-3 w-3'>
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  className='w-full h-full'
                  viewBox='0 0 37.286 37.287'
                  style={{ fill: '#F5385D', marginTop: 5 }}>
                  <g>
                    <path
                      d='M36.683,16.339l-7.567,7.377l1.786,10.417c0.128,0.75-0.182,1.509-0.797,1.957c-0.348,0.253-0.762,0.382-1.176,0.382
		c-0.318,0-0.638-0.076-0.931-0.23l-9.355-4.918l-9.355,4.918c-0.674,0.355-1.49,0.295-2.107-0.15
		c-0.615-0.448-0.924-1.206-0.795-1.957l1.787-10.417L0.604,16.34c-0.547-0.531-0.741-1.326-0.508-2.05
		c0.236-0.724,0.861-1.251,1.615-1.361l10.459-1.521l4.68-9.478c0.335-0.684,1.031-1.116,1.792-1.116
		c0.763,0,1.456,0.432,1.793,1.115l4.68,9.478l10.461,1.521c0.752,0.109,1.379,0.637,1.611,1.361
		C37.425,15.013,37.226,15.808,36.683,16.339z'
                    />
                  </g>
                </svg>
              </div>

              <p className='pl-1 text-sm'>{reviews_per_month}</p>
            </div>
          </div>
        </>
      );
    } else {
      return (
        <div className='md:flex md:flex-wrap md:items-center md:justify-start'>
          <p className='mt-3 text-sm sm:my-2 font-light text-gray-600'>
            {country}
          </p>
        </div>
      );
    }
  };

  return (
    <Link href='/stays/[id]' as={`/stays/${id}`}>
      <div className='cursor-pointer'>
        <div>
          <img className='w-full h-full rounded' src={picture_url} />
        </div>

        <div className=''>{renderhost(host_is_superhost)}</div>
        <div>
          <p
            style={{ fontFamily: 'airbnb-book' }}
            className='text-sm md:text-base text-gray-850 truncate'>
            {name}
          </p>
        </div>
        <div className='hidden md:block'>
          <p style={{ fontFamily: 'airbnb-book' }}>
            <span
              style={{ fontFamily: 'airbnb-bold' }}
              className='text-gray-850'>
              ${price} CAD
            </span>
            /night
          </p>
        </div>
      </div>
    </Link>
  );
};
