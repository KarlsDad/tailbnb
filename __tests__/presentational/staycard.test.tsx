import * as React from 'react';
import { shallow } from 'enzyme';

// Component
import { StayCard } from '../../src/components/functions/StayCard';

describe('StayCard', () => {
  it('should render correctly in "debug" mode', () => {
    const info = {
      id: 'testid',
      host_is_superhost: 'testhost_is_superhost',
      name: 'testname',
      price: 10,
      country: 'testcountry',
      reviews_per_month: 10,
      picture_url: 'testpicture_url'
    };
    const component = shallow(<StayCard {...info} />);
    expect(component).toMatchSnapshot();
  });
});
