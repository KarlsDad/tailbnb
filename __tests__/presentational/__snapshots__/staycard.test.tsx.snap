// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StayCard should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
